File: android/src/main/kotlin/tenko/rx_flutter_plugin/ObservableCallback.kt
package tenko.rx_flutter_plugin

enum class ObservableCallbackType {
    onNext, onComplete, onError
}

data class ObservableCallback(
        val observableCallbackType: ObservableCallbackType,
        val requestId: Int,
        val payload: Any?,
        val errorMessage: String?
) {

}

File: ios/Classes/ObservableCallback.swift
struct ObservableCallback {
    let callbackType: CallbackType
    let requestId: Int
    let payload: Any?
    let errorMessage: String?
    
    enum CallbackType: String {
        case ON_NEXT = "onNext"
        case ON_COMPLETE = "onComplete"
        case ON_ERROR = "onError"
    }
    
    func toDictionary() -> [String: Any?] {
        return [
            Field.OBSERVABLE_CALLBACK.rawValue: self.callbackType.rawValue,
            Field.REQUEST_ID.rawValue: self.requestId,
            Field.PAYLOAD.rawValue: self.payload,
            Field.ERROR_MESSAGE.rawValue: self.errorMessage
        ]
    }
}
